pkg/controller/csr/csr_controller.go

```diff
@@ -393,3 +491,17 @@ func (r *reconcileCSR) getPod(ctx context.Context, csr *certificatesv1.Certifica
 
 	return nil, nil
 }
+
+func (r *reconcileCSR) getHostEndpoint(ctx context.Context, hostname string) (*v3.HostEndpoint, error) {
+	hepList, err := r.calicoClient.ProjectcalicoV3().HostEndpoints().List(ctx, metav1.ListOptions{FieldSelector: fmt.Sprintf("spec.node=%s", hostname)})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil, nil
+		}
+		return nil, err
+	}
+	if len(hepList.Items) == 0 {
+		return nil, nil
+	}
```

Comment:
If for some reason the filter returns more items add a warning.

```suggestion
  if len(hepList.Items) > 1 {
        log.WithFields(log.Fields{
            "hostname": hostname,
            "count":    len(hepList.Items),
        }).Warn("Multiple HostEndpoints found for node; returning the first one")
    }
```